Get rid of compiler warnings

There are several warnings when compiling. Let's get rid of them:

Restoring NuGet packages...
To prevent NuGet from restoring packages during build, open the Visual Studio Options dialog, click on the Package Manager node and uncheck 'Allow NuGet to download missing packages during build.'
1>------ Build started: Project: Updater, Configuration: Debug x86 ------
2>------ Build started: Project: Models, Configuration: Debug x86 ------
3>------ Build started: Project: SWIMFrame, Configuration: Debug x86 ------
1>  Updater -> C:\Work\ApsimX\Bin\Updater.exe
3>  SWIMFrame -> C:\Work\ApsimX\Bin\SWIMFrame.exe
2>  Models -> C:\Work\ApsimX\Bin\Models.exe
4>------ Build started: Project: Importer, Configuration: Debug x86 ------
5>------ Build started: Project: APSIMRunner, Configuration: Debug x86 ------
5>  APSIMRunner -> C:\Work\ApsimX\Bin\APSIMRunner.exe
4>  Importer -> C:\Work\ApsimX\Models\Bin\Importer.exe
6>------ Build started: Project: ApsimNG, Configuration: Debug x86 ------
6>  **No way to resolve conflict between "System.Xml.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" and "System.Xml.Linq, Version=2.0.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, Retargetable=Yes". Choosing "System.Xml.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" arbitrarily.
6>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
6>C:\Work\ApsimX\ApsimNG\Views\ExplorerView.cs(741,17,741,43): warning CS0612: 'CellRenderer.CancelEditing()' is obsolete
6>C:\Work\ApsimX\ApsimNG\Presenters\TreeProxyPresenter.cs(41,26,41,39): warning CS0169: The field 'TreeProxyPresenter.soilMidpoints' is never used
6>C:\Work\ApsimX\ApsimNG\Views\ExplorerView.cs(221,45,221,63): warning CS0067: The event 'ExplorerView.ShortcutKeyPressed' is never used**
6>  ApsimNG -> C:\Work\ApsimX\Bin\ApsimNG.exe
7>------ Build started: Project: UnitTests, Configuration: Debug x86 ------
7>  **No way to resolve conflict between "DocumentFormat.OpenXml, Version=2.7.2.0, Culture=neutral, PublicKeyToken=8fb06cb64d019a17" and "DocumentFormat.OpenXml, Version=2.5.5631.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35". Choosing "DocumentFormat.OpenXml, Version=2.7.2.0, Culture=neutral, PublicKeyToken=8fb06cb64d019a17" arbitrarily.**
7>  UnitTests -> C:\Work\ApsimX\Tests\UnitTests\bin\Debug\UnitTests.dll
========== Build: 7 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========

--- a/ApsimNG/ApsimNG.csproj
+++ b/ApsimNG/ApsimNG.csproj
@@ -614,9 +614,6 @@
       <HintPath>..\packages\ExcelNumberFormat.1.0.3\lib\net20\ExcelNumberFormat.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="FastMember.Signed, Version=0.0.0.0, Culture=neutral, PublicKeyToken=9e8f22703bef9a29, processorArchitecture=MSIL">
-      <HintPath>..\packages\FastMember.Signed.1.2.0\lib\net45\FastMember.Signed.dll</HintPath>
-    </Reference>
     <Reference Include="gdk-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\DeploymentSupport\Windows\Bin64\gdk-sharp.dll</HintPath>
@@ -637,8 +634,9 @@
       <HintPath>..\DeploymentSupport\Windows\Bin64\gtkdotnet.dll</HintPath>
       <Private>False</Private>
     </Reference>
-    <Reference Include="HtmlAgilityPack, Version=1.6.16.0, Culture=neutral, PublicKeyToken=bd319b19eaf3b43a, processorArchitecture=MSIL">
-      <HintPath>..\packages\HtmlAgilityPack.1.6.16\lib\Net45\HtmlAgilityPack.dll</HintPath>
+    <Reference Include="HtmlAgilityPack, Version=1.6.17.0, Culture=neutral, PublicKeyToken=bd319b19eaf3b43a, processorArchitecture=MSIL">
+      <HintPath>..\packages\HtmlAgilityPack.1.6.17\lib\Net45\HtmlAgilityPack.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="ICSharpCode.NRefactory, Version=5.0.0.0, Culture=neutral, PublicKeyToken=d4bfe873e7598c49, processorArchitecture=MSIL">
       <HintPath>..\packages\ICSharpCode.NRefactory.5.5.1\lib\Net40\ICSharpCode.NRefactory.dll</HintPath>

--- a/ApsimNG/ApsimNG.csproj
+++ b/ApsimNG/ApsimNG.csproj
@@ -795,6 +795,7 @@
       <HintPath>..\packages\System.Spatial.5.8.3\lib\net40\System.Spatial.dll</HintPath>
     </Reference>
     <Reference Include="System.XML" />
+    <Reference Include="System.Xml.Linq" />
     <Reference Include="webkit-sharp, Version=1.1.15.0, Culture=neutral, PublicKeyToken=eaa1d335d2e19745, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>Assemblies\webkit-sharp.dll</HintPath>

--- a/ApsimNG/Interfaces/IExplorerView.cs
+++ b/ApsimNG/Interfaces/IExplorerView.cs
@@ -130,9 +130,6 @@ namespace UserInterface.Interfaces
         /// <summary>Invoked then a node is renamed.</summary>
         event EventHandler<NodeRenameArgs> Renamed;
 
-        /// <summary>Invoked when a shortcut key is pressed.</summary>
-        event EventHandler<KeysArgs> ShortcutKeyPressed;
-
         /// <summary>Refreshes the entire tree from the specified descriptions.</summary>
         /// <param name="nodeDescriptions">The nodes descriptions.</param>
         void Refresh(NodeDescriptionArgs nodeDescriptions);

--- a/ApsimNG/Presenters/ExplorerPresenter.cs
+++ b/ApsimNG/Presenters/ExplorerPresenter.cs
@@ -110,7 +110,6 @@ namespace UserInterface.Presenters
             this.view.AllowDrop += this.OnAllowDrop;
             this.view.Droped += this.OnDrop;
             this.view.Renamed += this.OnRename;
-            this.view.ShortcutKeyPressed += this.OnShortcutKeyPress;
 
             this.view.Refresh(this.GetNodeDescription(this.ApsimXFile));
             this.WriteLoadErrors();
@@ -125,7 +124,6 @@ namespace UserInterface.Presenters
             this.view.AllowDrop -= this.OnAllowDrop;
             this.view.Droped -= this.OnDrop;
             this.view.Renamed -= this.OnRename;
-            this.view.ShortcutKeyPressed -= this.OnShortcutKeyPress;
             this.HideRightHandPanel();
             if (this.view is Views.ExplorerView)
             {
@@ -849,16 +847,6 @@ namespace UserInterface.Presenters
             }
         }
 
-        /// <summary>User has pressed one of our shortcut keys.</summary>
-        /// <param name="sender">Event sender</param>
-        /// <param name="e">Event arguments</param>
-        private void OnShortcutKeyPress(object sender, KeysArgs e)
-        {
-            if (e.Keys == ConsoleKey.F5)
-            {
-                contextMenu.RunAPSIM(sender, null);
-            }
-        }
         #endregion
 
         #region Privates        

--- a/ApsimNG/Presenters/TreeProxyPresenter.cs
+++ b/ApsimNG/Presenters/TreeProxyPresenter.cs
@@ -36,11 +36,6 @@ namespace UserInterface.Presenters
         private PropertyPresenter propertyPresenter;
 
         /// <summary>
-        /// The mid points of the soil
-        /// </summary>
-        private double[] soilMidpoints;
-
-        /// <summary>
         /// Attach the presenter
         /// </summary>
         /// <param name="model">The model object</param>

--- a/ApsimNG/Views/ExplorerView.cs
+++ b/ApsimNG/Views/ExplorerView.cs
@@ -217,9 +217,6 @@ namespace UserInterface.Views
         /// <summary>Invoked then a node is renamed.</summary>
         public event EventHandler<NodeRenameArgs> Renamed;
 
-        /// <summary>Invoked when a global key is pressed.</summary>
-        public event EventHandler<KeysArgs> ShortcutKeyPressed;
-
         /// <summary>Refreshes the entire tree from the specified descriptions.</summary>
         /// <param name="nodeDescriptions">The nodes descriptions.</param>
         public void Refresh(NodeDescriptionArgs nodeDescriptions)
@@ -738,7 +735,7 @@ namespace UserInterface.Views
         private void OnDragBegin(object sender, DragBeginArgs e)
         {
             if (textRender.Editable) // If the node to be dragged is open for editing (renaming), close it now.
-                textRender.CancelEditing();
+                textRender.StopEditing(true);
             DragStartArgs args = new DragStartArgs();
             args.NodePath = SelectedNode; // FullPath(e.Item as TreeNode);
             if (DragStarted != null)

--- a/ApsimNG/packages.config
+++ b/ApsimNG/packages.config
@@ -6,7 +6,7 @@
   <package id="EntityFramework" version="6.2.0" targetFramework="net46" />
   <package id="ExcelNumberFormat" version="1.0.3" targetFramework="net46" />
   <package id="FastMember.Signed" version="1.2.0" targetFramework="net46" />
-  <package id="HtmlAgilityPack" version="1.6.16" targetFramework="net46" />
+  <package id="HtmlAgilityPack" version="1.6.17" targetFramework="net46" />
   <package id="ICSharpCode.NRefactory" version="5.5.1" targetFramework="net452" />
   <package id="ICSharpCode.TextEditor" version="3.2.1.6466" targetFramework="net45" />
   <package id="MarkdownDeep.NET" version="1.5" targetFramework="net452" />

--- a/Models/Models.csproj
+++ b/Models/Models.csproj
@@ -93,8 +93,9 @@
     <Reference Include="ICSharpCode.SharpZipLib, Version=0.86.0.518, Culture=neutral, PublicKeyToken=1b03e6acf1164f73, processorArchitecture=MSIL">
       <HintPath>..\packages\SharpZipLib.0.86.0\lib\20\ICSharpCode.SharpZipLib.dll</HintPath>
     </Reference>
-    <Reference Include="MathNet.Numerics, Version=3.20.2.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MathNet.Numerics.3.20.2\lib\net40\MathNet.Numerics.dll</HintPath>
+    <Reference Include="MathNet.Numerics, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\MathNet.Numerics.4.0.0\lib\net40\MathNet.Numerics.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="MigraDoc.DocumentObjectModel-gdi, Version=1.50.3638.0, Culture=neutral, PublicKeyToken=f94615aa0424f9eb, processorArchitecture=MSIL">
       <HintPath>..\packages\PDFsharp-MigraDoc-gdi.1.50.3638-beta\lib\net20\MigraDoc.DocumentObjectModel-gdi.dll</HintPath>

--- a/Models/packages.config
+++ b/Models/packages.config
@@ -2,7 +2,7 @@
 <packages>
   <package id="ExcelDataReader" version="3.4.0" targetFramework="net46" />
   <package id="ExcelDataReader.DataSet" version="3.4.0" targetFramework="net46" />
-  <package id="MathNet.Numerics" version="3.20.2" targetFramework="net46" />
+  <package id="MathNet.Numerics" version="4.0.0" targetFramework="net46" />
   <package id="PDFsharp-MigraDoc-gdi" version="1.50.3638-beta" targetFramework="net40" />
   <package id="SharpZipLib" version="0.86.0" targetFramework="net4" />
 </packages>

--- a/SWIMFrame/SWIMFrame.csproj
+++ b/SWIMFrame/SWIMFrame.csproj
@@ -64,8 +64,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="MathNet.Numerics, Version=3.20.2.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MathNet.Numerics.3.20.2\lib\net40\MathNet.Numerics.dll</HintPath>
+    <Reference Include="MathNet.Numerics, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\MathNet.Numerics.4.0.0\lib\net40\MathNet.Numerics.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />

--- a/SWIMFrame/packages.config
+++ b/SWIMFrame/packages.config
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="MathNet.Numerics" version="3.20.2" targetFramework="net46" />
+  <package id="MathNet.Numerics" version="4.0.0" targetFramework="net46" />
 </packages>

--- a/Tests/UnitTests/UnitTests.csproj
+++ b/Tests/UnitTests/UnitTests.csproj
@@ -84,8 +84,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="MathNet.Numerics, Version=3.20.2.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\MathNet.Numerics.3.20.2\lib\net40\MathNet.Numerics.dll</HintPath>
+    <Reference Include="MathNet.Numerics, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\MathNet.Numerics.4.0.0\lib\net40\MathNet.Numerics.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">
       <HintPath>..\..\packages\NUnit.3.9.0\lib\net45\nunit.framework.dll</HintPath>

--- a/Tests/UnitTests/packages.config
+++ b/Tests/UnitTests/packages.config
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="MathNet.Numerics" version="3.20.2" targetFramework="net46" />
+  <package id="MathNet.Numerics" version="4.0.0" targetFramework="net46" />
   <package id="NUnit" version="3.9.0" targetFramework="net46" />
   <package id="NUnit3TestAdapter" version="3.9.0" targetFramework="net45" />
 </packages>

